ui/pages/confirmations/components/confirm/info/approve/approve.tsx

```diff
@@ -1,223 +1,89 @@
-
-  const transactionMeta = useSelector(
-    currentConfirmationSelector,
-  ) as TransactionMeta;
-
-  const { tokenAmount, formattedTokenNum, value } = useApproveTokenSimulation(
-    transactionMeta,
-    decimals,
-  );
-
-  const SpendingCapElement = (
-    <ConfirmInfoRowText
-      text={
-        tokenAmount === UNLIMITED_MSG
-          ? `${t('unlimited')} ${tokenSymbol}`
-          : `${formattedTokenNum} ${tokenSymbol}`
-      }
-      onEditClick={
-        transactionMeta.type === TransactionType.tokenMethodIncreaseAllowance
-          ? () => console.log('TODO on a following ticket')
-          : undefined
-      }
-      editIconClassName="edit-spending-cap"
-    />
-  );
-
-  if (!value) {
-    return null;
-  }
-
-  return (
-    <>
-      <ConfirmInfoRowDivider />
-
-      <ConfirmInfoRow
-        label={t('spendingCap')}
-        tooltip={t('spendingCapTooltipDesc')}
-      >
-        {tokenAmount === UNLIMITED_MSG ? (
-          <Tooltip title={formattedTokenNum}>{SpendingCapElement}</Tooltip>
-        ) : (
-          SpendingCapElement
-        )}
-      </ConfirmInfoRow>
-    </>
-  );
-};
+  const [isOpenEditSpendingCapModal, setIsOpenEditSpendingCapModal] =
+    useState(false);
+  const [customSpendingCap, setCustomSpendingCap] = useState('');
 
-const SpendingCap = () => {
-  const t = useI18nContext();
-
-  const transactionMeta = useSelector(
-    currentConfirmationSelector,
-  ) as TransactionMeta;
-
-  const { userBalance, tokenSymbol, decimals } = useAssetDetails(
-    transactionMeta.txParams.to,
-    transactionMeta.txParams.from,
-    transactionMeta.txParams.data,
+  const { decimals } = useAssetDetails(
+    transactionMeta?.txParams?.to,
+    transactionMeta?.txParams?.from,
+    transactionMeta?.txParams?.data,
   );
 
-  const accountBalance = new BigNumber(userBalance || '0')
-    .dividedBy(new BigNumber(10).pow(Number(decimals || '0')))
-    .toNumber();
-
-  const { pending } = useApproveTokenSimulation(
+  const customTxParamsData = useMemo(() => {
+    return getCustomTxParamsData(transactionMeta?.txParams?.data, {
+      customPermissionAmount: customSpendingCap || '0',
+      decimals,
+    });
+  }, [customSpendingCap, transactionMeta?.txParams?.data, decimals]);
+
+  const { value: estimatedGasLimit } = useAsyncResult(async () => {
+    return await estimateGas({
+      from: transactionMeta.txParams.from,
+      to: transactionMeta.txParams.to,
+      value: transactionMeta.txParams.value,
+      data: customTxParamsData,
+    });
+  }, [customTxParamsData, customSpendingCap, decimals]);
+
+  const [shouldUpdateConfirmation, setShouldUpdateConfirmation] =
+    useState(false);
+  const updateConfirmation = useCallback(() => {
+    if (shouldUpdateConfirmation && estimatedGasLimit) {
+      transactionMeta.txParams.data = customTxParamsData;
+      transactionMeta.txParams.gas = hexToDecimal(estimatedGasLimit as string);
+
+      dispatch(updateCurrentConfirmation(transactionMeta));
```

Comment:
Are we not deriving the confirmation now via the context? Meaning the Redux state is no longer used?

Reply:
That's true but we are currently in an "in between" state, I believe the changes to the submission flow will be done in [one of the next PRs](https://github.com/MetaMask/metamask-extension/pull/26685/files#diff-4935af77d52052112ed0a8efa32c2b8f63ceeb55cfa0704eb27ca3878033ccf7L101)

Reply:
I updated this now, as per https://github.com/MetaMask/metamask-extension/pull/26906#discussion_r1745111513